pages/app-layout/runtime-drawers-with-updates.page.tsx

```diff
@@ -0,0 +1,196 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useContext, useState } from 'react';
+import {
+  AppLayout,
+  ContentLayout,
+  Header,
+  HelpPanel,
+  Drawer,
+  Link,
+  SpaceBetween,
+  SplitPanel,
+  Toggle,
+} from '~components';
+import appLayoutLabels from './utils/labels';
+import { AppLayoutProps } from '~components/app-layout';
+import { Breadcrumbs, Containers } from './utils/content-blocks';
+import './utils/external-widget';
+import AppContext, { AppContextType } from '../app/app-context';
+import awsuiPlugins from '~components/internal/plugins';
+
+type DemoContext = React.Context<
+  AppContextType<{
+    hasTools: boolean | undefined;
+    hasDrawers: boolean | undefined;
+    splitPanelPosition: AppLayoutProps.SplitPanelPreferences['position'];
+  }>
+>;
+
+export default function WithDrawers() {
+  const [activeDrawerId, setActiveDrawerId] = useState<string | null>(null);
+  const [helpPathSlug, setHelpPathSlug] = useState<string>('default');
+  const { urlParams, setUrlParams } = useContext(AppContext as DemoContext);
+  const hasTools = urlParams.hasTools ?? false;
+  const hasDrawers = urlParams.hasDrawers ?? true;
+  const [isToolsOpen, setIsToolsOpen] = useState(false);
+  const [showBadge, setShowBadge] = useState(false);
+  const [turnOffResizable, setTurnOffResizable] = useState(false);
+
+  const drawersProps: Pick<AppLayoutProps, 'activeDrawerId' | 'onDrawerChange' | 'drawers'> | null = !hasDrawers
```

Comment:
This feature is irrelevant for the update functionality. I recommend removing this and related code to keep the demo simple and more focused on the use-case

Reply:
updated.